module/zfs/vdev_trim.c

```diff
@@ -1435,6 +1435,52 @@ vdev_autotrim_restart(spa_t *spa)
 		vdev_autotrim(spa);
 }
 
+/*
+ * A wrapper which calls vdev_trim_ranges(). It is intended to be called
+ * on leaf vdevs.
+ */
+int
+vdev_trim_simple(vdev_t *vd, uint64_t start, uint64_t size, trim_type_t type)
+{
+	trim_args_t		ta;
+	range_seg64_t 		physical_rs;
+	int			error;
+	physical_rs.rs_start = start;
+	physical_rs.rs_end = start + size;
+
+	ASSERT(vdev_is_concrete(vd));
+	ASSERT(vd->vdev_ops->vdev_op_leaf);
+
+	ta.trim_vdev = vd;
+	ta.trim_tree = range_tree_create(NULL, RANGE_SEG64, NULL, 0, 0);
+	ta.trim_type = type;
+	ta.trim_extent_bytes_max = zfs_trim_extent_bytes_max;
+	ta.trim_extent_bytes_min = zfs_trim_extent_bytes_min;
```

Comment:
Since we're requesting that a single extent be trimmed where the caller provides the size I don't think we want to apply the `zfs_trim_extent_bytes_min` limit.  Otherwise, this function cannot be used to TRIM ranges <=32k in size (with the defaults).  Similar to what's done for secure TRIM in `vdev_trim_thread()`, I'd suggest setting this to `SPA_MINBLOCKSIZE`.

`ta.trim_extend_bytes_min = SPA_MINBLOCKSIZE;`

This isn't really an issue for the L2ARC caller, but we may find this helper useful elsewhere in the future.